ouroboros-consensus/src/Ouroboros/Consensus/Ledger/Byron.hs

```diff
@@ -861,6 +864,25 @@ mkByronTx tx = ByronTx
       -- cardano-ledger#581
     }
 
+{-------------------------------------------------------------------------------
+  Block Fetch integration
+-------------------------------------------------------------------------------}
+
+byronBlockOrEBBMatchesHeader :: Header (ByronBlockOrEBB cfg)
+                             -> ByronBlockOrEBB cfg
+                             -> Bool
+byronBlockOrEBBMatchesHeader blkOrEbbHdr blkOrEbb =
+    case unByronHeaderOrEBB blkOrEbbHdr of
+        Left _ebbHdr -> isEbb (unByronBlockOrEBB blkOrEbb)
+        Right hdr    -> isBlockAndHeaderMatchesBody (unByronBlockOrEBB blkOrEbb) hdr
+  where
+    isEbb (CC.Block.ABOBBoundary _) = True
+    isEbb (CC.Block.ABOBBlock _)    = False
```

Comment:
So we're not doing any body check for the EBBs. The old code probably does. However I think this is actually fine. We'll not cause a loss of consensus with the old nodes by being more permissive here. And of course the new code never generates any.

That said, I think it's worth calling this fact out here as a comment in the code.

Reply:
I skipped the EBB body check because, in the `cardano-ledger` code, the `cardano-ledger` EBB types (`ABoundaryHeader` and `ABoundaryBody`) don't seem to provide enough information for me to validate whether a EBB header matches a EBB body.

I'll ask someone on the ledger team about what the plans are for these EBB data types. If there are plans to expand upon the EBB data types, it'd be worth opening a ticket which will remind me to revisit this.

Reply:
Also, as you mentioned, there is a header-body validation process for EBBs in the old code which is basically just verifying the hash of the list of slot leaders (`verifyGenesisBlock`). However, this isn't modeled in `cardano-ledger`.

Reply:
Added a comment.

Reply:
From @nc6 in relation to whether we'll be performing EBB proof checks in `cardano-ledger`:

> I think it's probably unnecessary. Let's consider what happens if we don't validate. Then somebody could (in theory) pass us an EBB with a fake body attached. But what do we do with that body? We throw it away. All subsequent blocks are checked against the correct headers anyway, so it can't cause any cascade issues